test/GroupsTests.jl

```diff
@@ -235,4 +235,18 @@ gx["e"] = BenchmarkGroup([], "1" => g["e"]["1"][x["a"]], "3" => g["e"]["3"][x["c
 
 @test g[x] == gx
 
+# deep index assignment in BenchmarkGroup #
+#-----------------------------------------#
```

Comment:
Maybe change this heading to `indexing by Vector` to match the other headers (which in turn match the headers in the manual)